pkg/controller/defaults/memory.go

```diff
@@ -0,0 +1,83 @@
+package defaults
+
+import (
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	adminv1 "github.com/acorn-io/acorn/pkg/apis/internal.admin.acorn.io/v1"
+	"github.com/acorn-io/baaah/pkg/router"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// defaultMemory calculates the default that should be used and considers the defaults from the Config, ComputeClass, and
+// runtime ComputeClass
+func addDefaultMemory(req router.Request, cfg *apiv1.Config, appInstance *v1.AppInstance) error {
+	if appInstance.Status.Defaults.Memory == nil {
+		appInstance.Status.Defaults.Memory = v1.MemoryMap{}
+	}
+
+	var (
+		defaultWC string
+		err       error
+	)
+	if value, ok := appInstance.Spec.ComputeClass[""]; ok {
+		defaultWC = value
+	} else {
+		defaultWC, err = adminv1.GetDefaultComputeClass(req.Ctx, req.Client, appInstance.Namespace)
+		if err != nil {
+			return err
+		}
+	}
+
+	appInstance.Status.Defaults.Memory[""] = cfg.WorkloadMemoryDefault
+	wc, err := adminv1.GetAsProjectComputeClassInstance(req.Ctx, req.Client, appInstance.Status.Namespace, defaultWC)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	if wc != nil {
+		parsedMemory, err := adminv1.ParseComputeClassMemory(wc.Memory)
+		if err != nil {
+			return err
+		}
+		def := parsedMemory.Def.Value()
+		appInstance.Status.Defaults.Memory[""] = &def
+	}
+
+	if err := addWorkloadMemoryDefault(req, appInstance, cfg.WorkloadMemoryDefault, appInstance.Status.AppSpec.Containers); err != nil {
+		return err
+	}
+
+	if err := addWorkloadMemoryDefault(req, appInstance, cfg.WorkloadMemoryDefault, appInstance.Status.AppSpec.Jobs); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addWorkloadMemoryDefault(req router.Request, appInstance *v1.AppInstance, configDefault *int64, containers map[string]v1.Container) error {
+	for name, container := range containers {
+		memory := configDefault
+		computeClass, err := adminv1.GetClassForWorkload(req.Ctx, req.Client, appInstance.Spec.ComputeClass, container, name, appInstance.Namespace)
+		if computeClass == nil && err != nil {
```

Comment:
Are there any situations when we would return a `computeClass` and an `err`, but want to continue with this function?

Reply:
So the only ways that GetClassForWorkload will fail is if:

1. The Client is failing to list ComputeClasses
2. The ComputeClass that was defined for a workload does not exist

In either situation I would say we should fail out the controller route chain. Thoughts?

Reply:
My question is should this just be `if err != nil`? Is there any reason to continue if `computeClass != nil` and `err != nil`?

Reply:
Ah I see what you're saying now. I believe this is a left-over from a previous implementation of the code base that no longer exists. In other words - yes, this should just be `if err != nil`